libnixf/src/Basic/diagnostic.py

```diff
@@ -184,11 +184,35 @@ class Diagnostic(TypedDict):
         "message": "undefined variable `{}`",
     },
     {
-        "sname": "sema-def-not-used",
-        "cname": "DefinitionNotUsed",
-        "severity": "Hint",
+        "sname": "sema-unused-def-generic",
```

Comment:
Wait, seems this line is not covered by any test case. I suppose this code is actually unreachable because other definitions will not raise that warning?